### PR TITLE
feat: Use cloudevent for alerting endpoint

### LINKED ISSUE #266: Refactoring: Use cloudevents.WithGetHandlerFunc(HTTPGetHandler) for alerting endpoint
At the moment we open two ports for prometheus-service:

One port that we use for retrieving alerts from prometheus directly, and a second one for CloudEvents.

https://github.com/keptn-contrib/prometheus-service/blob/a41d9c0893974a884ae94f8296b5944c84a6f43f/main.go#L79-L103

With `cloudevents.WithGetHandlerFunc(HTTPGetHandler)` we could easily work around that, e.g.:

```golang
 	p, err := cloudevents.NewHTTP(cloudevents.WithPath(env.Path), cloudevents.WithPort(env.Port), cloudevents.WithGetHandlerFunc(HTTPGetHandler))

...

// HTTPGetHandler will handle all requests for '/health' and '/ready'
func HTTPGetHandler(w http.ResponseWriter, r *http.Request) {
	switch r.URL.Path {
	case "/health":
		healthEndpointHandler(w, r)
	case "/ready":
		healthEndpointHandler(w, r)
	default:
		endpointNotFoundHandler(w, r)
	}
}
```

# Definition of Done

- [x] HTTP Handling reduced to one port (8080), and uses clouevents http handler instead of `http.HandleFunc`
- [x] env config updated to reflect this change
- [x] Helm Chart updated

### PR DESCRIPTION
**Pull request by TannerGabriel**
## This PR

- Use `cloudevents.WithGetHandlerFunc()` function to implement health checks and alerting endpoint  
- Remove other HTTP Endpoints
- Change `PORT` environment variable

## Fixes
#266

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -80,6 +61,14 @@ func gotEvent(event cloudevents.Event) error {
 	var shkeptncontext string
 	_ = event.Context.ExtensionAs("shkeptncontext", &shkeptncontext)
 
+	logger := keptncommon.NewLogger(shkeptncontext, "", utils.ServiceName)
+
+	// Send prometheus alert
+	if event.SpecVersion() == "" {
+		eventhandling.ProcessAndForwardAlertEvent(event.Data(), logger, shkeptncontext)
+		return nil
+	}
```
Comment (christian-kreuzberger-dtx) [inline main.go:70]: I think this won't work :) I understand the reasoning behind it though, and code-wise it looks nice.

Anyway, here is why this won't work:

Prometheus itself sends the alert to prometheus-service. This alert does **not** send a proper CloudEvent, also it does not send an shkeptncontext, nor a specversion. I think we will never get into `gotEvent(event cloudevents.Event)` in that case.
An example payload is available in prometheus_alert.json in this repo (I will add a helper tool to execute an HTTP request against localhost:8080 with that payload in another PR).

Anyway, long story short, I tried it out using
```
POST http://localhost:8080/
Accept: application/json
Cache-Control: no-cache
Content-Type: application/json

< ./prometheus_alert.json
```

and I get 
```
Response code: 415 (Unsupported Media Type); Time: 287ms; Content length: 0 bytes
```

This is what I expected, we never get into `gotEvent`, because the CloudEvents library is blocking us here.

**Review by christian-kreuzberger-dtx — CHANGES_REQUESTED** [review]
As mentionedin the code comment, this doesn't work as expected.

But we are on the right path with this change.

I think what we could do is this:

```
 	p, err := cloudevents.NewHTTP(cloudevents.WithPath(env.Path), cloudevents.WithPort(env.Port), cloudevents.WithGetHandlerFunc(HTTPGetHandler))
```
* Change env.Path to `/event`
* Handle `POST /` requests in `HTTPGetHandler` - those are the ones coming from Prometheus; forward those to `ProcessAndForwardAlertEvent`.

File: main.go (hunk 1 above)
Comment (christian-kreuzberger-dtx) [reply main.go:70]: Related PR: https://github.com/keptn-contrib/prometheus-service/pull/271

**Comment by TannerGabriel** [comment]
> 

Setting the `env.Path` variable to `/event` will also affect the `HTTPGetHandler` with the prefix which would result in the following endpoints:

- `/health` -> `/event/health`
- `/ready` -> `/event/ready`
- `/` -> `/event`

Since this would lead to no response from the handler I think we need to reconsider our options here. You are saying that the `POST /` requests are from Prometheus. Then which path is used for the `cloudevents`?

**Comment by TannerGabriel** [comment]
@christian-kreuzberger-dtx Also, would it be possible to change the path of the received Prometheus alerts to something like `/prometheus` instead of `/`?

**Comment by christian-kreuzberger-dtx** [comment]
@TannerGabriel and myself discussed this in a Zoom Call. We don't want to rename the prometheus endpoint, as this would be a breaking change would require everyone to re-configure Prometheus Alert Manager.

Other then that, there seems to be a limitation with CloudEvents SDK Go, which we can bypass by using `http.Handle` directly.

**Review by christian-kreuzberger-dtx — APPROVED** [review]
thanks for making those changes, LGTM

also tested it locally, works fine :)
